test/test_nn.py

```diff
@@ -3271,7 +3271,7 @@ def _test_InstanceNorm_general(self, cls, input, device="cpu", dtype=torch.float
 
     def _test_InstanceNorm_cuda_half(self, cls, input):
         # THNN
-        input = Variable(input.cuda().half().random_(1, 10), requires_grad=True)
+        input = input.cuda().half().random_(1, 10).requires_grad_(True)
```

Comment:
You missed this line

Reply:
It looks like this line is overwriting the input tensor called `input`. Not sure how to change this as we're not explicitly constructing a tensor here, would we want to do something like `input = torch.tensor(input.clone().detach(), device="cuda", dtype=torch.half).random_(1,10).requires_grad_(True)`?

Reply:
Let's do `input.to(device='cuda', dtype='half').random_(...)...` to get it all done in one go.